cmd/apiserver/main.go

```diff
@@ -2,202 +2,23 @@ package main
-	for _, calcName := range calcNameList {
-		logger := logrus.WithFields(logrus.Fields{"calculation": calcName, "worker": workerName})
-		o.simulateRun(ctx, fakeClient, calcName, logger)
-	}
-}
-
-func (o *options) simulateRun(ctx context.Context, fakeClient v1.VegaV1Interface, calcName string, logger *logrus.Entry) {
-	logger.Info("Starting simulation")
-
-	ticker := time.NewTicker(time.Duration(o.dryTickerMinutes) * time.Minute)
-	defer ticker.Stop()
-	done := make(chan bool)
-
-	go func() {
-		opts := metav1.SingleObject(metav1.ObjectMeta{Name: calcName})
-		watcher, _ := fakeClient.Calculations().Watch(ctx, opts)
-		defer watcher.Stop()
-
-		// Watch calculation until Completed or Failed status
-		for {
-			select {
-			case event, _ := <-watcher.ResultChan():
-				obj := event.Object.(*calculationsv1.Calculation)
-				if obj.Phase == calculationsv1.CompletedPhase || obj.Phase == calculationsv1.FailedPhase {
-					done <- true
-				}
-			}
-		}
-	}()
-
-	getPhaseWithFailureChance := func(chance int) calculationsv1.CalculationPhase {
-		r := rand.Intn(100)
-		if r < chance {
-			return calculationsv1.FailedPhase
-		}
-		return calculationsv1.CompletedPhase
-	}
-
-	isFinished := func(spec calculationsv1.CalculationSpec) bool {
-		for _, step := range spec.Steps {
-			if step.Status == calculationsv1.ProcessingPhase || step.Status == calculationsv1.CreatedPhase {
-				return false
-			}
-		}
-		return true
-	}
-
-	for {
-		select {
-		case <-done:
-			logger.Warn("Simulation finished")
-			return
-		case <-ticker.C:
-			newCalc, err := fakeClient.Calculations().Get(ctx, calcName, metav1.GetOptions{})
-
-			if err != nil {
-				logger.WithError(err).Error("couldn't get calculation")
-				goto End
-			}
-
-			switch newCalc.Phase {
-			case calculationsv1.CreatedPhase:
-				newCalc.Phase = calculationsv1.ProcessingPhase
-				newCalc.Spec.Steps[0].Status = calculationsv1.ProcessingPhase
-				break
-
-			case calculationsv1.ProcessingPhase:
-				if isFinished(newCalc.Spec) {
-					newCalc.Phase = calculationsv1.CompletedPhase
-				}
-
-				for index, step := range newCalc.Spec.Steps {
-					switch step.Status {
-
-					case calculationsv1.CompletedPhase:
-						continue
-
-					case calculationsv1.FailedPhase:
-						newCalc.Phase = calculationsv1.FailedPhase
-						goto End
-
-					case calculationsv1.ProcessingPhase:
-						newCalc.Spec.Steps[index].Status = getPhaseWithFailureChance(o.dryRunFailureRate)
-						goto End
-
-					case calculationsv1.CreatedPhase:
-						newCalc.Spec.Steps[index].Status = calculationsv1.ProcessingPhase
-						goto End
-					}
-				}
-			}
-
-		End:
-			logger.Info("Updating calculation")
-			fakeClient.Calculations().Update(ctx, newCalc, metav1.UpdateOptions{})
-		}
-	}
-}
-
 func main() {
-	o := gatherOptions()
+	sim := gatherOptions()
 
-	ctx, cancel := context.WithCancel(context.Background())
+	var cancel context.CancelFunc
+	sim.ctx, cancel = context.WithCancel(context.Background())
```

Comment:
this is wrong. We already create a context. You should pass this in simulator when we run with dry run as well.

Reply:
Do we need to have the context with cancel or is `context.Background()` enough?

Reply:
You don't need to change this at all. It's not related with the changes you want to introduce in this PR.